### PR TITLE
include versions like '1.18.x' in the k8s dropdown

### PR DESCRIPTION
**Pull request by laverya**

### CODE REVIEW

File: src/components/Kurlsh.js [hunk 1]
```diff
@@ -836,6 +847,31 @@ class Kurlsh extends React.Component {
     }
   }
 
+  // add versions like "1.19.x" to the list of installable versions
+  addK8sVersions(actualVersions) {
```
Comment (jgruica) [inline src/components/Kurlsh.js:851]: i'll prefer this function to be arrow function.
also you can change code inside if you want to follow that pattern too:

```suggestion
  addK8sVersions = (actualVersions) => {
    // get a list of the distinct minor versions
    const minorVersionsRegex = /^1\.[0-9]+/g;
    let minorVersions = []

    actualVersions.forEach(v => {
      const matches = v.version.match(minorVersionsRegex);
      if (matches && matches.length === 1 && !minorVersions.includes(matches[0])) {
        minorVersions.push(matches[0])
      }
    })

    // for each minor version, find the first version in the actualVersions array that matches
    // and insert `1.minor.x` before it
    minorVersions.forEach(mv => {
      const isMatch = actualVersions.find(av => av.version.startsWith(mv));
      if (!!isMatch) {
        actualVersions.splice(actualVersions.indexOf(isMatch), 0, {version: mv+".x"});
      }
    })

    return actualVersions
  }
```

File: src/components/Kurlsh.js [hunk 2]
```diff
@@ -591,8 +591,19 @@ class Kurlsh extends React.Component {
 
   getLabel = name => ({ version }) => {
     if (version === "latest") {
-      const latest = this.state.versions[name][1];
-      version = `latest (${latest.version})`;
+      if (name === "kubernetes") {
```
Comment (jgruica) [inline src/components/Kurlsh.js:594]: i'll prefer this to be written as a separate function and called inside this one.

``` 
 generateVersionLabel = (name, version) => {
    if (version === "latest") {
      if (name === "kubernetes") {
        const latest = this.state.versions[name][2]; // for k8s, the first version is a ".x" version
        version = `latest (${latest.version})`;
      } else {
        const latest = this.state.versions[name][1];
        version = `latest (${latest.version})`;
      }
    } else if (version.endsWith(".x")) {
      const versionIndex = this.state.versions[name].findIndex((element) => element.version === version);
      if (this.state.versions[name].length > versionIndex) { // if there is a member of the array after the one specified
        const next = this.state.versions[name][versionIndex+1]
        version = `${version} (${next.version})`
      }
    } 
    return version;
  }
  ```
  
  and getLabel after looks more cleaner:
  ```
    getLabel = name => ({ version }) => {
    const generatedVersion = this.generateVersionLabel(name, version);
    return (
      <div className="versionLabel--wrapper">
        <span className="versionLabel" style={{ fontSize: 14 }}>{generatedVersion}</span>
      </div>
    );
  }
  ```

File: src/components/Kurlsh.js [hunk 3]
```diff
@@ -591,8 +591,19 @@ class Kurlsh extends React.Component {
...
+        const latest = this.state.versions[name][2]; // for k8s, the first version is a ".x" version
+        version = `latest (${latest.version})`;
+      } else {
+        const latest = this.state.versions[name][1];
+        version = `latest (${latest.version})`;
+      }
+    } else if (version.endsWith(".x")) {
```
Comment (jgruica) [inline src/components/Kurlsh.js:601]: does this also needs a check is name kubernetes?

Comment (laverya) [reply src/components/Kurlsh.js:601]: I'll put it in for now

**Review by jgruica — APPROVED** [review]

**Pull request merged**